### PR TITLE
Adopt fork of express-prometheus-middleware

### PR DESCRIPTION
**Pull request by grantforrester-nes**
## Proposed Change Summary
- Added github configuration
- Removed config files for tools not used at NES
- Reworked path normalization to address deprecation warning
- Fixed liniting errors
- Removed vulnerable collection of gc stats
- Removed unused authentication feature
- Upgraded to prom-client 14
- Updated documentation esp. example code
- Added unit tests

Reviewing by commit recommended.

## Evidence
Evidence of changes / testing

I create a little app ...

```js
const express = require('express')
const promMid = require('@nes-digital-service/express-prometheus-middleware')
const promClient = require('prom-client')

const API_PORT = 8080
const TELEMETRY_PORT = 9090

const app = express()
app.use(promMid({
  collectDefaultMetrics: true,
  requestDurationBuckets: [0.1, 0.5, 1, 1.5],
  requestLengthBuckets: [512, 1024, 5120, 10240, 51200, 102400],
  responseLengthBuckets: [512, 1024, 5120, 10240, 51200, 102400],
}));

// curl -X GET localhost:8081/hello?name=Chuck%20Norris
app.get('/hello', (req, res) => {
  console.log('GET /hello')
  const { name = 'Anon' } = req.query
  res.json({ message: `Hello, ${name}!` })
});

app.listen(API_PORT, () => {
  console.log(`Example api is listening on http://localhost:${API_PORT}/hello`);
});

const telemetry = express()

// curl -X GET localhost:9091/metrics
telemetry.get('/metrics', async (req, res) => {
  console.log('GET /metrics')
  res.set('Content-Type', promClient.register.contentType)
  res.end(await promClient.register.metrics())
})

telemetry.listen(TELEMETRY_PORT, () => {
  console.log(`Telemetry is listening on http://localhost:${TELEMETRY_PORT}/metrics`);
});
```

Fired a few requests at it and then pulled the metrics....

```shell
$ curl http://localhost:8080/hello?name=Chuck
{"message":"Hello, Chuck!"}
...
$ curl http://localhost:9090/metrics
# HELP http_request_duration_seconds Duration of HTTP requests in seconds
# TYPE http_request_duration_seconds histogram
http_request_duration_seconds_bucket{le="0.1",route="/hello",method="GET",status="2XX"} 5
http_request_duration_seconds_bucket{le="0.5",route="/hello",method="GET",status="2XX"} 5
http_request_duration_seconds_bucket{le="1",route="/hello",method="GET",status="2XX"} 5
```

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2018 João Paulo Fontenele Brito
+Copyright (c) 2022 NES Digital Service
```
Comment (Kris-Mcalpine) [inline LICENSE:4]: suggestion: I think we need to leave the original copyright notice here  according to the license but can add our own underneath. Some related discussions here: https://opensource.stackexchange.com/questions/4439/how-to-deal-with-licences-after-forking-a-project

**Review by Kris-Mcalpine — CHANGES_REQUESTED** [review]
Just one small suggestion Grant, but other than that, looks good. The other teams (and us) are going to find this very handy!

**Comment by grantforrester-nes** [comment]
I have amended the original commit to update the licence as suggested. Cheers. 👍

**Review by Kris-Mcalpine — APPROVED** [review]
LGTM! 👍

**Pull request merged**